resources/lib/addon.py

```diff
@@ -215,11 +224,24 @@ def play(item_to_play):
 
 
 def clearcache():
-    shutil.rmtree(tempdir, True)
-    addon = xbmcaddon.Addon(id=__addonid__)
-    heading = addon.getLocalizedString(32004)
-    message = addon.getLocalizedString(32005)
-    xbmcgui.Dialog().notification(heading, message, xbmcgui.NOTIFICATION_INFO, 1000)
+    xbmc.log(f"Cache is being cleared", level=xbmc.LOGDEBUG)
+    try:
+        # delete the channels.xml
+        channel_file_path = os.path.join(tempdir, "channels.xml")
+        if os.path.isfile(channel_file_path):
+            os.remove(channel_file_path)
+        # delete all folders
+        for lsItem in os.listdir(tempdir):
```

Comment:
What difference does this solve? It will still remove everything in the tempdir, just not remove the folder itself?

Reply:
It preserves the settings.xml file. Not sure if there is a better place for caching the channels or if they should all be in their own subdirectoy to make clearing the cache easier.

Reply:
Feedback from here: https://github.com/xbmc/repo-plugins/pull/3804#discussion_r775150161

Reply:
Ahh I get you - so it'll only remove `channels.xml` and all directories within that dir.